### PR TITLE
Tock 2.0: Remove LegacyDriver

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

Now that capsules have switched, this removes LegacyDriver.

### Testing Strategy

travis

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: kernel/src/sched.rs [hunk 1]
```diff
@@ -838,29 +838,13 @@ impl Kernel {
...
-                            ))
-                        } else {
-                            GenericSyscallReturnValue::Legacy(d.subscribe(
-                                subdriver_number,
-                                None,
-                                process.appid(),
-                            ))
-                        }
-                    }
                     None => GenericSyscallReturnValue::Legacy(ReturnCode::ENOSUPPORT),
```
Comment (hudson-ayers) [inline kernel/src/sched.rs:848]: It seems like we probably don't still want the `Legacy` variant here?
```suggestion
                    None => callback.into_subscribe_failure(ErrorCode::NOSUPPORT),
```

File: kernel/src/sched.rs [hunk 2]
```diff
@@ -884,24 +868,12 @@ impl Kernel {
...
-                    }
+                    Some(d) => GenericSyscallReturnValue::from_command_result(d.command(
+                        subdriver_number,
+                        arg0,
+                        arg1,
+                        process.appid(),
+                    )),
                     None => {
                         // System call transition note: This does not
                         // match the expected error code for the Tock
```
Comment (hudson-ayers) [inline kernel/src/sched.rs:879]: This comment should be removed

File: kernel/src/sched.rs [hunk 3]
```diff
@@ -935,24 +907,11 @@ impl Kernel {
...
-                                    ld.allow_readwrite(process.appid(), subdriver_number, oslice)
-                                }
-                                Err(err) => err, /* memory not valid */
-                            };
-
-                            GenericSyscallReturnValue::Legacy(rc)
-                        }
                         None => {
                             // System call transition note: This does
                             // not match the expected error code for
```
Comment (hudson-ayers) [inline kernel/src/sched.rs:917]: This comment is also stale

File: kernel/src/sched.rs (hunk 1 above)
Comment (hudson-ayers) [reply kernel/src/sched.rs:848]: Once this is done, I think the variant can also be removed from syscall.rs

**Comment by bradjc** [comment]
Updated to remove old comments and legacy return type.

**Review by hudson-ayers — APPROVED** [review]

**Review by ppannuto — APPROVED** [review]
🥳

**Review by lschuermann — APPROVED** [review]

**Pull request merged**